### PR TITLE
[ROCM] rocm 6.0 fixes followup

### PR DESCRIPTION
**Pull request by pemeliya**
This is a follow-up PR for rocm-6.0 platform support.
I have also adapted two unit tests which were failing on specific architectures / platform version.

@xla-rotation: would you take a look, please?

### CODE REVIEW

**Review by ezhulenev — APPROVED** [review]

File: xla/stream_executor/device_description.h [hunk 1]
```diff
@@ -159,33 +159,52 @@ class RocmComputeCapability {
...
+    return gfx9_mi200_or_later();
   }
 
+  bool fence_before_barrier() const {
+    return gfx_version() != "gfx900" && gfx_version() != "gfx906";
+  }
+
+  bool has_hipblaslt() const {
+    return gfx9_mi200_or_later();
+  };
```
Comment (akuegel) [inline xla/stream_executor/device_description.h:206]: The linter is complaining about the ";" character in this line.
Also it is complaining that there is no newline after the last line in this file.

**Review by akuegel — CHANGES_REQUESTED** [review]

**Review by akuegel — APPROVED** [review]

**Comment by pemeliya** [comment]
@ddunl JAX Linux GPU build seems to fail due to some configuration problem.. perhaps the build should be restarted?

**Comment by ddunl** [comment]
Not sure what happened on those last runs, I hadn't seen that before, looks good now that it re-ran

**Comment by pemeliya** [comment]
@ddunl : it seems that not all changes have been merged, particularly, I don't see these ones in the upstream:
https://github.com/openxla/xla/pull/7277/files#diff-710053e487634d28f9fbe5c4ee69dae088a90bc525d2e3cb8af81dabf640fc27R89-R90

https://github.com/openxla/xla/pull/7277/files#diff-710053e487634d28f9fbe5c4ee69dae088a90bc525d2e3cb8af81dabf640fc27R89-R90

Could it be that because these changes reside in third_party subdirectory, they were missed somehow ?

**Comment by ddunl** [comment]
That's likely, I'll have a change rectifying this today. Thanks for pointing this out!!

**Pull request closed without merging**